### ISSUE TITLE
Failed to load next.config.js (Windows)

### ISSUE THREAD

**Issue by devwilliamy** [muxinc/next-video#151]
Hi,

I'm running into problems with Vercel blob.

I've tried integrating it with a project I'm already working on and also from a project made from scratch and I'm running into the same error.

This is my next.config.js:

```
const { withNextVideo } = require('next-video/process');
/** @type {import('next').NextConfig} */
const nextConfig = {};

module.exports = withNextVideo(nextConfig, {
    provider: "vercel-blob",
  });
 ```

I'm on Next 14.0.4 (project I'm working on) and Next 14.1.0 (barebone project)

I've run `npx next-video init` and it's able to detect that I have a `next.config.js` 

I have a 17 MB video hosted on Vercel blob.

This is how I'm using it in a `page.tsx` (I took the mux promo vercel blob):

```
import NextVideo from 'next-video';
import getStarted from 'https://p2kzwwhnykpzlhrx.public.blob.vercel-storage.com/videos/mux-promo-2DDmlWCcJkJ2ZQxWcWtj6nVwsM4n1z.mp4';

export default function Page() {
return <NextVideo src={getStarted} />;
}
```

I'm getting this when I go to the page (the video plays but I don't want that warning to show up):
![image](https://github.com/muxinc/next-video/assets/156253900/ac39cfca-7002-4e9a-96f9-561392d4a43c)

When I try to run `npx next-video sync` it says 

```
Failed to load next.config.js or next.config.mjs
✗ An unknown error occurred MuxError: The MUX_TOKEN_ID environment variable is missing or empty; either provide it, or instantiate the Mux client with an tokenId option, like new Mux({ tokenId: 'my tokenId' }).
```

I put in my env variables into `.env.development.local` and then when I run it again, I get:

```
Failed to load next.config.js or next.config.mjs
✗ An unknown error occurred Error: Missing video `folder` config.
at getAssetConfigPath (file:///C:/git/vercel-blob-client/node_modules/next-video/dist/assets.js:63:13)
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
    at async updateAsset (file:///C:/git/vercel-blob-client/node_modules/next-video/dist/assets.js:47:21)
    at async uploadRequestedFile (file:///C:/git/vercel-blob-client/node_modules/next-video/dist/providers/mux/provider.js:180:27)

    at async Promise.all (index 2)
    at async Object.handler (file:///C:/git/vercel-blob-client/node_modules/next-video/dist/cli/sync.js:86:24)
```

And after looking through issues I've never seen that one before so I'm a little lost.

Are there hidden folders I don't know about? Why is my next.config.js not being recognized?

I saw issue #119 but I cleared out my videos folder and I never tried uploading to mux on my fresh project. Wasn't sure if I should have continued the conversation there or start a new one.

**Comment by luwes** [comment]
have a look at this Codesandbox here and let me know if you can't get it to work like that.
https://codesandbox.io/p/devbox/next-video-vercel-blob-w6wwqw?file=%2Fapp%2Fpage.js%3A14%2C17

generally you would import from the local `/videos/` folder. 
next-video takes care of the upload with running `npx next-video sync`.

```js
import NextVideo from "next-video";
import awesomeVideo from "/videos/mux-promo.mp4";

export default function IndexPage() {
  return (
    <>
      <h1>
        <a href="https://github.com/muxinc/next-video" target="_blank">
          <code>next-video</code>
        </a>
      </h1>
      <NextVideo
        src={awesomeVideo}
        style={{
          aspectRatio: 16 / 9,
        }}
      />
    </>
  );
}
```

**Comment by devwilliamy** [comment]
Hi @luwes , I've downloaded the code sandbox and tried adding my video into the video folder. This is what shows up:
![image](https://github.com/muxinc/next-video/assets/156253900/56b5bd55-fcfd-4ef1-a1a8-57ff2facb6a3)

This is the .mp4.json:
```
{"status":"sourced",
"originalFilePath":"videos\\file_example_MP4_1920_18MG.mp4",
"provider":"vercel-blob",
"providerMetadata":{},
"createdAt":1706639506952,
"updatedAt":1706639506952,
"size":17839845}
```

I've also added `BLOB_READ_WRITE_TOKEN` to my `.env.local` and also a `.env.development.local` and I still get this error:
```
PS C:\Users\...\next-video-vercel-blob> npx next-video sync
(node:37584) [DEP0040] DeprecationWarning: The `punycode` module is deprecated. Please use a userland alternative instead.
(Use `node --trace-deprecation ...` to show where the warning was created)
Failed to load next.config.js or next.config.mjs
✗ An unknown error occurred MuxError: The MUX_TOKEN_ID environment variable is missing or empty; either provide it, or instantiate the Mux client with an tokenId option, like new Mux({ tokenId: 'my tokenId' }).
    at new Mux (file:///C:/Users/.../next-video-vercel-blob/node_modules/@mux/mux-node/index.mjs:29:19)
    at initMux (file:///C:/Users/.../next-video-vercel-blob/node_modules/next-video/dist/providers/mux/provider.js:12:17)
    at Module.uploadLocalFile (file:///C:/Users/.../next-video-vercel-blob/node_modules/next-video/dist/providers/mux/provider.js:100:3)
    at uploadLocalFile (file:///C:/Users/.../next-video-vercel-blob/node_modules/next-video/dist/handlers/local-upload.js:6:23)
    at callHandler (file:///C:/Users/.../next-video-vercel-blob/node_modules/next-video/dist/video-handler.js:11:10)
    at existingFileProcessor (file:///C:/Users/.../next-video-vercel-blob/node_modules/next-video/dist/cli/sync.js:74:16)
    at async Promise.all (index 0)
    at async Object.handler (file:///C:/Users/.../next-video-vercel-blob/node_modules/next-video/dist/cli/sync.js:86:24)
```

After I add the mux env's, it will then try to upload the asset to mux. Is this something that's intended? Although I do notice that the provider does say vercel blob now... 

So does it still get uploaded on Mux but it'll get it from vercel blob somehow? Could you help me understand what's going on here?

```
{"status":"uploading",
"originalFilePath":"videos\\file_example_MP4_1920_18MG.mp4",
"provider":"vercel-blob",
"providerMetadata":
{"mux":{"uploadId":"QekAYtn7vJe7AdiPfgkPmM0269Fler6lpnJ85nRM01374"}},
"createdAt":1706639506952,
"updatedAt":1706639938735,
"size":17839845}
```

**Comment by luwes** [comment]
yes, seems like it's not able to load next.config.js and that leads to other problems like loading the wrong provider.

there is a next.config.js file in your `next-video-vercel-blob` folder right?

**Comment by devwilliamy** [comment]
Yes, it is as you had it in the code sandbox.

![image](https://github.com/muxinc/next-video/assets/156253900/72667f7a-7167-4186-a453-1154f2a8315f)

```
// @ts-check
const { withNextVideo } = require("next-video/process");

/** @type {import('next').NextConfig} */
const nextConfig = {};

module.exports = withNextVideo(nextConfig, {
  provider: "vercel-blob",
});

````

Something I also noticed was the json looked different than the mux-promo one.

Mine:
```
{"status":"ready",
"originalFilePath":"videos\\file_example_MP4_1920_18MG.mp4",
"provider":"vercel-blob",
"providerMetadata":{"mux":{"uploadId":"QekAYtn7vJe7AdiPfgkPmM0269Fler6lpnJ85nRM01374",
"assetId":"5PlO4y8STI02W02J702hoIcytlNAVnLe7502ClBUKJPe9BE",
"playbackId":"CWaE00p9oyCNIj01nmRPZr9nWopOIacgkjA87J00A11c34"}},
"createdAt":1706639506952,
"updatedAt":1706640460594,
"size":17839845,
"blurDataURL":"data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAACAAAAASCAYAAAA6yNxSAAAJcklEQVR4AQCBAH7/...//SXl+O+3YTiAAAAAElFTkSuQmCC"}
```

Mux Promo:
```
{
  "status": "ready",
  "originalFilePath": "videos/mux-promo.mp4",
  "provider": "vercel-blob",
  "providerMetadata": {
    "vercel-blob": {
      "url": "https://p2kzwwhnykpzlhrx.public.blob.vercel-storage.com/videos/mux-promo-2DDmlWCcJkJ2ZQxWcWtj6nVwsM4n1z.mp4",
      "contentType": "video/mp4"
    }
  },
  "createdAt": 1703101538992,
  "updatedAt": 1703101706894,
  "size": 15328870,
  "sources": [
    {
      "src": "https://p2kzwwhnykpzlhrx.public.blob.vercel-storage.com/videos/mux-promo-2DDmlWCcJkJ2ZQxWcWtj6nVwsM4n1z.mp4",
      "type": "video/mp4"
    }
  ]
}

```

**Comment by devwilliamy** [comment]
Hi @luwes , it _appears_ to be a Windows vs MacOS issue?

I took the next-video-vercel-blob to my Mac and it was able to read the next.config.js and upload to Vercel Blob with no issues. I even took my original project that I was having trouble with and it was able to upload to Mux or Vercel Blob. I have the same config on both sides and at first I thought maybe something in my project could have been interfering but it worked just fine on my Macbook.

**Comment by luwes** [comment]
this should be fixed in v0.12.1! https://github.com/muxinc/next-video/pull/159

could you give it a go?
thanks!

**Comment by devwilliamy** [comment]
Hi @luwes thanks for the quick turn-around. I'm no getting the issues on my Windows machine & I've confirmed I can upload to Vercel Blob.
